docs/operating/cross-cluster-tls-communication.md

```diff
@@ -0,0 +1,398 @@
+                  cluster: sidecar_name
+                  host_rewrite_literal: thanos.sidecardomain.com
+          http_filters:
+          - name: envoy.filters.http.router
+  clusters:
+  - name: sidecar_name
+    connect_timeout: 30s
+    type: LOGICAL_DNS
+    http2_protocol_options: {}
+    dns_lookup_family: V4_ONLY
+    lb_policy: ROUND_ROBIN
+    load_assignment:
+      cluster_name: sidecar_name
+      endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                socket_address:
+                  address: thanos.sidecardomain.com
+                  port_value: 443
+    transport_socket:
+      name: envoy.transport_sockets.tls
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
+        common_tls_context:
+          tls_certificates:
+            - certificate_chain:
+                filename: /certs/tls.crt
+              private_key:
+                filename: /certs/tls.key
+          validation_context:
+            trusted_ca:
+              filename: /certs/cacerts.pem
+          alpn_protocols:
+          - h2
+          - http/1.1
+        sni: thanos.sidecardomain.com
+
+```
+### `service.yaml`
+This is the service for the envoy sidecar. You will need to define a new port for every sidecar you would like to add.
+
+```
+kind: Service
+apiVersion: v1
+metadata:
+  name: thanos-global-test-envoy
+  namespace: thanos-global
+  labels:
+    name: thanos-global-test-envoy
+spec:
+  ports:
+    - name: [port_name]
+      protocol: TCP
+      port: 10000
+      targetPort: 10000
+    - name: [port_name_2]
+      protocol: TCP
+      port: 10001
+      targetPort: 10001
+  selector:
+    name: thanos-global-test-querier
+  type: ClusterIP
+  sessionAffinity: None
+```
+
+### `ingress.yaml`
+This is an example ingress for a remote sidecar using NGINX ingress. You must use TLS (port 443 - limitation from NGINX) as HTTP2 is only supported on a separate listener (see [here](https://github.com/kubernetes/ingress-nginx/issues/3938))
+
+You must have certs configured and the CA added into the envoy sidecar earlier to allow verification (if using client cert v3 envoy config)
+
+```
+kind: Ingress
+apiVersion: extensions/v1beta1
+metadata:
+  name: monitoring-rancher-monitor-thanos-gateway
+  namespace: monitoring
+  annotations:
+    nginx.ingress.kubernetes.io/backend-protocol: GRPC
+    nginx.ingress.kubernetes.io/force-ssl-redirect: 'true'
+    nginx.ingress.kubernetes.io/grpc-backend: 'true'
+    nginx.ingress.kubernetes.io/protocol: h2c
+    nginx.ingress.kubernetes.io/proxy-read-timeout: '160'
+spec:
+  tls:
+    - hosts:
+        - thanos.sidecardomain.com
+      secretName: thanos-tls
+  rules:
+    - host: thanos.sidecardomain.com
+      http:
+        paths:
+          - path: /
+            backend:
+              serviceName: monitoring-rancher-monitor-prometheus
+              servicePort: 10901
+```
```

Comment:
What about some `## Summary` section here? :hugs

Reply:
It would be nice what is End State of the setup, what you can do if you deploy those on kubernetes to sum up. (: